tools/bench/functional/plot_permutations.py

```
@@ -0,0 +1,163 @@
+    ax.set_title(f"threshold={thresh}")
+    ax.set_xlabel("tile_size")
+    ax.set_xticks(tile_sizes)
+    ax.grid(True, alpha=0.3)
+
+axes[0].set_ylabel("ppp time (ns)")
+axes[-1].legend(title="tag_buffer", bbox_to_anchor=(1.05, 1), loc="upper left")
+plt.tight_layout()
+plt.savefig("perms_ppp_lines.png", dpi=120, bbox_inches="tight")
+print("saved perms_ppp_lines.png")
+
+# ── Figure 2: heatmaps ppp_time [tile_size × tag_buffer] per threshold ──
```

Ruff (0.15.15)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)